### DISCUSSION TITLE
Template (like issue templates) for PR

### DISCUSSION THREAD

**Discussion by Torxed** [community/community#27897] [Pull Requests]
Similar topics:
* https://github.com/orgs/community/discussions/2838
* https://github.com/orgs/community/discussions/4527

It would be nice with some drop downs in the PR that the contributor could use.
As well as check boxes to ask the contributor if they've read the contribution guidelines, have they run local tests, etc.

**Comment by Borda** [comment]
Hi, this may be rather a Feed topic than PR :otter: 
so creating [one](https://github.com/orgs/community/discussions/42973) there, but happy to drop it later... :rabbit:

**Comment by jrhemstad** [comment]
Big upvote from me on this as well.

Issue forms were a game changer for us to provide more structured input, require fields, etc.

We really want to be able to do the same thing with pull request templates. For example, I want to be able to require a linked issue for a PR.

**Reply by Torxed** [reply]
Exactly, currently it's very hard to make PR'a look roughly the same.

Even a `[ ] I have tested the code` checkbox does some difference, but currently most people just remove it from the PR text with Ctrl+A -> DEL and make their own PR text.

Having similar PR's would help to streamline info gathering.

• https://github.com/archlinux/archinstall/pull/1815
• https://github.com/archlinux/archinstall/pull/1805
• https://github.com/archlinux/archinstall/pull/1575 <- What I've tried to dictate

All so different :)

**Comment by JonathanRenon-EDB** [comment]
I think it could also help with integration testing, by running more specific integration test depending on the answer to the questions.
things like:
- is the change OS specific ? 
- platform specific ? 
- for CLI tools: does it impact a specific argument ? 
- ensuring mentions to linked tickets are present

all that in a generalized format which helps reviewers analyze PR more efficiently.
